File: Data Structures & Algorithms/pow-x-n/submission-2.cs
public class Solution {
    public double MyPow(double x, int n) {
        var res = x;
        if (n > 0) {
            while (n-- > 1) {
                res *= x;
            }
        } else if (n < 0) {
            res = 1.0 / MyPow(x, -n);
        } else {
            return 1;
        }

        return res;
    }
}

File: Data Structures & Algorithms/mergeSort/submission-2.py
# Definition for a pair.
# class Pair:
#     def __init__(self, key: int, value: str):
#         self.key = key
#         self.value = value
class Solution:
    def mergeSort(self, pairs: List[Pair]) -> List[Pair]:

        if len(pairs) <= 1:
            return pairs

        pivot = len(pairs) // 2
        left_partition = self.mergeSort(pairs[:pivot])
        right_partition = self.mergeSort(pairs[pivot:])

        return self.merge(left_partition, right_partition)

    def merge(self, left: List[Pair], right: List[Pair]) -> List[Pair]:
        res = []
        while left and right:
            if left[0].key <= right[0].key:
                res.append(left.pop(0))
            else:
                res.append(right.pop(0))
        
        while left:
            res.append(left.pop(0))
        
        while right:
            res.append(right.pop(0))
        return res
